ic-cdk/src/call.rs

```diff
@@ -565,9 +567,16 @@ unsafe extern "C" fn callback<T: AsRef<[u8]>>(state_ptr: *const RwLock<CallFutur
     if let Some(state) = state.upgrade() {
         // Make sure to un-borrow_mut the state.
         {
-            state.write().unwrap().result = Some(match RejectCode::from(msg_reject_code()) {
-                RejectCode::NoError => Ok(msg_arg_data()),
-                c => Err(CallError::CallRejected(c, msg_reject_msg())),
+            state.write().unwrap().result = Some(match msg_reject_code() {
+                0 => Ok(msg_arg_data()),
+                code => {
+                    let reject_code = RejectCode::try_from(code).unwrap();
```

Comment:
this unwrap could be dangerous for canisters with an outdated CDK version if the replica introduces a new reject code